moveit_ros/planning/plan_execution/src/plan_execution.cpp

```diff
@@ -477,18 +477,22 @@ void plan_execution::PlanExecution::doneWithTrajectoryExecution(
 void plan_execution::PlanExecution::successfulTrajectorySegmentExecution(const ExecutableMotionPlan* plan,
                                                                          std::size_t index)
 {
-  ROS_DEBUG("Completed '%s'", plan->plan_components_[index].description_.c_str());
-
   // if any side-effects are associated to the trajectory part that just completed, execute them
-  if (plan->plan_components_[index].effect_on_success_)
-    if (!plan->plan_components_[index].effect_on_success_(plan))
-    {
-      // execution of side-effect failed
-      ROS_ERROR("Execution of path-completion side-effect failed. Preempting.");
-      preempt_requested_ = true;
-      return;
+    if(plan->plan_components_.size() != 0) {
```

Comment:
Could you please change the logic to
```
if(plan->plan_components_.size() == 0){
  ... report warning ...
  return;
}
```
?
Multiple MoveIt developers agreed that this is the preferred style in changed MoveIt code.